### DISCUSSION TITLE
How can i get message publisher？

### DISCUSSION THREAD

**Discussion by Alexety** [nats-io/nats.java#970] [Q&A] [answered]
How can I distinguish which leaf node is sending this message in the core cluster in Java. My architecture is a central cluster with multiple leaf clusters.

**Comment by scottf** [comment, marked as answer]
When you connect to a server with the java client, you get an instance of a `Connection` object. The `Connection` object has a method called `getServerInfo()`. `ServerInfo` has an id you can read with `getServerId()`. This id is always set by the server itself. There is also a name you can read with `getServerName()`. The name you can set yourself in the server conf with the `server_name` property

You could use this id or name as either part of your message payload, or as message header.

**Reply by Alexety** [reply]
Thank you for your reply. I think I have found a way to solve this problem.
